ui/app/src/components/PreviewComponentsPanel/PreviewComponentsPanel.tsx

```diff
@@ -114,22 +114,60 @@ export const ComponentsPanelUI: React.FC<ComponentsPanelUIProps> = (props) => {
 
   const onBrowseSharedInstancesClicked = () => {
     dispatch(
-      browseSharedInstance({
-        contentType: menuContext.contentType.id
-      })
+      batchActions([
+        browseSharedInstance({
+          contentType: menuContext.contentType.id
+        }),
+        pushToolsPanelPage({
+          id: 'craftercms.component.ToolsPanelPage',
+          configuration: {
+            title: 'previewBrowseComponentsPanel.title',
+            widgets: [
+              {
+                id: 'craftercms.component.PreviewBrowseComponentsPanel'
+              }
+            ]
+          }
+        })
+      ])
```

Comment:
Let's move the logic of using a the ToolsPanelPage to an util function that receives configuration `{ title: '', widgets: [...] }` and returns this whole chunk.

```js
pushToolsPanelPage(createToolsPanelPage({
  title: 'previewBrowseComponentsPanel.title',
  widgets: [...]
}))
```